akka-http-tests/src/main/java/akka/http/javadsl/server/examples/petstore/PetStoreExample.java

```diff
@@ -6,46 +6,55 @@
 
 import akka.actor.ActorSystem;
 import akka.http.javadsl.marshallers.jackson.Jackson;
+import akka.http.javadsl.model.StatusCodes;
 import akka.http.javadsl.server.*;
-import akka.http.javadsl.server.values.PathMatcher;
-import akka.http.javadsl.server.values.PathMatchers;
 
 import java.io.IOException;
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
 
 import static akka.http.javadsl.server.Directives.*;
+import static akka.http.javadsl.server.StringUnmarshaller.*;
 
 public class PetStoreExample {
-  static PathMatcher<Integer> petId = PathMatchers.intValue();
-  static RequestVal<Pet> petEntity = RequestVals.entityAs(Jackson.jsonAs(Pet.class));
 
+  private static Route putPetHandler(Map<Integer, Pet> pets, Pet thePet) {
+      pets.put(thePet.getId(), thePet);
+      return complete(StatusCodes.OK, thePet, Jackson.<Pet>json());
+  }
+    
   public static Route appRoute(final Map<Integer, Pet> pets) {
     PetStoreController controller = new PetStoreController(pets);
 
-    final RequestVal<Pet> existingPet = RequestVals.lookupInMap(petId, Pet.class, pets);
-
-    Handler1<Pet> putPetHandler = (Handler1<Pet>) (ctx, thePet) -> {
-      pets.put(thePet.getId(), thePet);
-      return ctx.completeAs(Jackson.json(), thePet);
+    // Defined as Function in order to refere to [pets], but this could also be an ordinary method.
+    Function<Integer, Route> existingPet = petId -> {
+        Pet pet = pets.get(petId);
+        return (pet == null) ? reject() : complete(StatusCodes.OK, pet, Jackson.<Pet>json());
     };
-
+      
     return
       route(
-        path().route(
+        path("", () ->
           getFromResource("web/index.html")
         ),
-        path("pet", petId).route(
-          // demonstrates three different ways of handling requests:
-
-          // 1. using a predefined route that completes with an extraction
-          get(extractAndComplete(Jackson.<Pet>json(), existingPet)),
+        pathPrefix("pet", () -> 
+          path(INT(), petId -> route(
+            // demonstrates different ways of handling requests:
 
-          // 2. using a handler
-          put(handleWith1(petEntity, putPetHandler)),
+            // 1. using a Function
+            get(() -> existingPet.apply(petId)),
 
-          // 3. calling a method of a controller instance reflectively
-          delete(handleReflectively(controller, "deletePet", petId))
+            // 2. using a method
+            put(() -> 
+              entityAs(Jackson.jsonAs(Pet.class), thePet -> 
```

Comment:
Here are a few alternatives:
1. `entity(Jackson.unmarshaller(Pet.class), thePet -> ...)`
2. `entity(Jackson.fromJson(Pet.class), thePet -> ...)`
3. `entityUsing(Jackson.unmarshaller(Pet.class), thePet -> ...)`
4. `entityUsing(Jackson.to(Pet.class), thePet -> ...)`

@ktoso which do you like best?

I'll rename the marshalling part of `Jackson` appropriately to what we choose here.

In all cases, I think I'll drop the variant that takes a `Class` as first argument. It's leftover from Scala where an unmarshaller is pulled in implicitly by type. Dropping it simplifies the DSL, and I think it's good to have a word ("json", "xml", "xmlOrJson") between the type and the http entity.